Issue #344: App Redesign: Transaction failed, submitted, and confirmation modals

jahabeebs opened the pull request:
Closes #344 

## Description
- Redesigns transaction failed, submitted, and confirmation modals
- The designs were slightly too large for mobile so I reduced the width of the modal but kept the larger font sizes. If you think it's better to have the text not wrap as much then I'll reduce the font sizes

## Screenshots

Transaction failed desktop

<img width="1378" alt="failed desktop" src="https://github.com/open-dollar/od-app/assets/47253537/dd772ff5-4178-41d4-988e-8e9d9ba1aadb">

Transaction failed mobile

<img width="447" alt="failed mobile" src="https://github.com/open-dollar/od-app/assets/47253537/e88ddba9-4c2d-409a-9447-08fff98f33d2">

Transaction submitted desktop

<img width="1408" alt="submitted desktop" src="https://github.com/open-dollar/od-app/assets/47253537/5661929e-7aba-4335-90ac-e50f446d3e35">

Transaction submitted mobile

<img width="458" alt="submitted mobile" src="https://github.com/open-dollar/od-app/assets/47253537/ce1d4094-95e1-4b78-8699-bc2d6bd1a5aa">

Transaction confirmation desktop

<img width="1452" alt="creating facilitator" src="https://github.com/open-dollar/od-app/assets/47253537/3e73f75f-ed3d-47d7-8492-e64edad4c52d">


Transaction confirmation mobile

<img width="496" alt="create facilitator mobile" src="https://github.com/open-dollar/od-app/assets/47253537/7aa4286b-c817-4c2a-a61a-300e1d73529e">

pi0neerpat reviewed (changes requested):
- Remove floating "x" to close, and add ability to click-away to close

- Reduce font and button size to increase padding
![image](https://github.com/open-dollar/od-app/assets/35622595/3e705666-07ad-4daa-bc8c-4641a3c8f6b2)

jahabeebs commented:
> * Remove floating "x" to close, and add ability to click-away to close
> 
>     * Reduce font and button size to increase padding
>       ![image](https://private-user-images.githubusercontent.com/35622595/322075742-3e705666-07ad-4daa-bc8c-4641a3c8f6b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMTY5OTgsIm5iZiI6MTcxMzIxNjY5OCwicGF0aCI6Ii8zNTYyMjU5NS8zMjIwNzU3NDItM2U3MDU2NjYtMDdhZC00ZGFhLWJjOGMtNDY0MWEzYzhmNmIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE1VDIxMzEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlYTU0NmVlNjVmNjVjMzUwM2Q0NTgwOWNmY2I1ZDMwZDgwMjIwZTRlNWI1MDk4Y2ZiYjY4NjI5YmEzNTlkNWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.i2JlHfhVlLU1zXwZ7uXLUO287svPKsOHlMNRVMeGmQU)

I made the following changes
- Decreased header font size which then decreased button size since they're in a flex container
- Added more padding to the sides but total modal width stays the same
- Got rid of X button in top right of modal and replaced it with being able to click out of the container to close it 

## Screenshots

Transaction failed

<img width="438" alt="smaller transaction failed" src="https://github.com/open-dollar/od-app/assets/47253537/5a90f38d-2e3f-48c3-8276-81f92adeaad6">

Click to close out of modal

https://github.com/open-dollar/od-app/assets/47253537/6a6a2777-4bbe-4fc5-9bb4-443be102d041

Transaction submitted 

<img width="741" alt="transaction submitted" src="https://github.com/open-dollar/od-app/assets/47253537/e216310d-8921-4178-a92d-67197253b124">

Waiting for confirmation 

<img width="582" alt="waiting for confirmation" src="https://github.com/open-dollar/od-app/assets/47253537/c44bf35e-a77a-4a7a-a7bc-f134b45bb3f3">
